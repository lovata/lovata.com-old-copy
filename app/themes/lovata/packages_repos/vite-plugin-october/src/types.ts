import type { SSROptions, ViteDevServer } from 'vite'

export interface ServerConfiguration {
	default: keyof ServerConfiguration['configs']
	aliases: Record<string, string>
	configs: Record<string, ViteConfiguration>
	commands: CommandsConfiguration
}

export interface CommandsConfiguration {
	artisan: Record<string, string[]> | string[]
	shell: Record<string, string[]> | string[]
}

export interface ViteConfiguration {
	entrypoints: {
		paths: string | string[]
		ssr?: string
		ignore?: string | string[]
	}
	build_path: string
	dev_server: {
		enabled?: boolean
		url: string
		cert?: string
		key?: string
	}
	commands?: CommandsConfiguration
	env_prefixes?: string[]
}

export type ResolvedConfiguration = ViteConfiguration & {
	configName?: string
	aliases: Record<string, string>
}

export interface Options {
	/**
	 * Path to PHP executable.
	 */
	php?: string

	/**
	 * A configuration object or a path to a configuration file.
	 * Setting to false disables reading the configuration file path from the `CONFIG_PATH_VITE` environment variable.
	 */
	config?: ResolvedConfiguration | string | false

	/**
	 * Post CSS plugins.
	 */
	postcss?: any[]

	/**
	 * SSR-specific options.
	 */
	ssr?: SSROptions

	/**
	 * Whether to automatically update the tsconfig.json file with aliases.
	 *
	 * @deprecated Use `vite.commands.artisan` => `vite:tsconfig` instead.
	 */
	updateTsConfig?: boolean

	/**
	 * Whether to allow overrides from the base configuration. If false, base
	 * options will be ignored, so stuff like `--host 0.0.0.0` won't work.
	 *
	 * @default true
	 */
	allowOverrides?: boolean

	/**
	 * List of file changes to listen to.
	 */
	watch?: WatchInput[] | WatchOptions
}

export interface WatchOptions {
	reloadOnBladeUpdates?: boolean
	reloadOnConfigUpdates?: boolean
	input?: WatchInput[]
}

export interface WatchInputHandlerParameters {
	file: string
	server: ViteDevServer
}

export interface WatchInput {
	condition: (file: string) => boolean
	handle: (parameters: WatchInputHandlerParameters) => void
}

export interface Certificates {
	key: string
	cert: string
}

export interface PhpFinderOptions {
	/**
	 * Actual path to PHP. This will be used instead of
	 */
	path?: string

	/**
	 * Custom environment variables.
	 */
	env?: any

	/**
	 * Either `production` or `development`. Used for loading the environment.
	 */
	mode?: string
}
