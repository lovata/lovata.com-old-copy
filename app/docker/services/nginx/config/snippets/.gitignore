*
!.gitignore
!whitelist.conf
!php.conf
!static.conf
!graphql.conf
!mailhog.conf
!gzip.conf
!redirects.conf
