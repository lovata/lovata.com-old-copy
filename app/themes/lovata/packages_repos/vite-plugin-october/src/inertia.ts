/**
 * Resolves a page component.
 */
export async function resolvePageComponent(name: string, pages: Record<string, any>, defaultLayout?: any) {
	const path = Object.keys(pages)
		.sort((a, b) => a.length - b.length)
		.find((path) => path.endsWith(`${name.replaceAll('.', '/')}.vue`))

	if (!path) {
		throw new Error(`Page component "${name}" could not be found.`)
	}

	let component = typeof pages[path] === 'function'
		? await pages[path]()
		: pages[path]

	component = component.default ?? component
	component.layout ??= defaultLayout

	return component
}
