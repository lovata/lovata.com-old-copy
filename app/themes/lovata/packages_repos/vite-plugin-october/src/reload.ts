import { Plugin, ViteDevServer } from 'vite'
import makeDebugger from 'debug'
import { warn } from './utils'
import { Options, WatchOptions } from './types'

const PREFIX = 'vite:laravel:reload'
const debug = makeDebugger(PREFIX)

/**
 * Reload when some files are changed.
 */
export const reload = (options: Options = {}): Plugin => {
	const watchOptions: Required<WatchOptions> = {
		input: [],
		reloadOnBladeUpdates: true,
		reloadOnConfigUpdates: true,
		...(Array.isArray(options.watch) ? { input: options.watch } : options.watch),
	}

	debug('Given options:', options)
	debug('Resolved options:', watchOptions)

	// When the config change, we want a full module graph
	// invalidation as well as a full reload
	if (watchOptions.reloadOnConfigUpdates) {
		watchOptions.input.push({
			condition: (file) => file.endsWith('config/vite.php'),
			handle: ({ server }) => {
				debug('Configuration file changed, invalidating module graph and reloading')
				server.moduleGraph.invalidateAll()
				server.ws.send({ type: 'full-reload', path: '*' })
			},
		})
	}

	// Blade files
	if (watchOptions.reloadOnBladeUpdates) {
		watchOptions.input.push({
			condition: (file) => file.endsWith('.htm'),
			handle: ({ server }) => {
				debug('Htm file changed, reloading')
				server.ws.send({ type: 'full-reload', path: '*' })
			},
		})
	}

	function handleReload(file: string, server: ViteDevServer) {
		file = file.replaceAll('\\', '/')

		watchOptions.input.forEach((value) => {
			if (value.condition(file)) {
				debug(`${file} changed, applying its handler`)
				try {
					value.handle({ file, server })
				} catch (error: any) {
					warn(PREFIX, `Handler failed for ${file}: ${error.message}`)
					debug('Full error:', error)
				}
			}
		})
	}

	return {
		name: 'vite:laravel:reload',
		configureServer(server) {
			server.watcher
				.on('add', (path) => handleReload(path, server))
				.on('change', (path) => handleReload(path, server))
				.on('unlink', (path) => handleReload(path, server))
		},
	}
}
