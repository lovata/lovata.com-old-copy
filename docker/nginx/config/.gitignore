*
!.gitignore
!snippets
!snippets/*
!snippets/staging/*
!snippets/production/*
!.htpasswd
