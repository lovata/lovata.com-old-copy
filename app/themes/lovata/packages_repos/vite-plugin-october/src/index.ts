import { PluginOption, UserConfig } from 'vite'
import { manifest } from './manifest'
import { config } from './config'
import { reload } from './reload'
import { callArtisan, callShell, findPhpPath } from './utils'
import type { Options } from './types'

export function defineConfig(base: UserConfig = {}) {
	return <UserConfig> {
		...base,
		plugins: [
			...base?.plugins as any,
			config(),
			reload(),
			manifest(),
		],
	}
}

export const laravel = (options: Options = {}): PluginOption[] => [
	reload(options),
	config(options),
	manifest(),
]

export * from './inertia'
export { manifest, reload, config, callArtisan, callShell, findPhpPath, Options }
export default laravel
