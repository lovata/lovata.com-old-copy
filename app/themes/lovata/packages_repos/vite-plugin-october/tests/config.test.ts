import { it, expect } from 'vitest'
import { readConfig } from '../src/config'
import { artisan, fixture } from './utils'

it('reads config from the given option object', () => {
	const result = readConfig({
		config: {
			aliases: { '@': 'assets' },
			build_path: 'build',
			dev_server: { url: 'http://localhost:5173' },
			entrypoints: { paths: 'assets/scripts/js' },
		},
	}, process.env)

	expect(result).toMatchSnapshot()
})

it('reads config from the given path', () => {
	const result = readConfig({ config: fixture('config.json') }, process.env)

	expect(result)
		.toMatchSnapshot()
})

it('reads config from the CONFIG_PATH_VITE environment variable', () => {
	process.env.CONFIG_PATH_VITE = fixture('config.json')

	expect(readConfig({}, process.env))
		.toMatchSnapshot()
})

it('does not read config the CONFIG_PATH_VITE if config is set to false', () => {
	process.env.TEST_ARTISAN_SCRIPT = artisan
	process.env.CONFIG_PATH_VITE = fixture('config.json')

	expect(readConfig({ config: false }, process.env))
		.toMatchSnapshot()
})

it('tries to read the specified configuration from php', () => {
	process.env.TEST_ARTISAN_SCRIPT = artisan

	expect(readConfig({ config: 'artisan' }, process.env))
		.toMatchSnapshot()
})

it('throws when the named configuration is not found', () => {
	process.env.TEST_ARTISAN_SCRIPT = artisan

	expect(() => readConfig({ config: 'named-but-not-defined' }, process.env))
		.toThrow('Could not read configuration')
})
