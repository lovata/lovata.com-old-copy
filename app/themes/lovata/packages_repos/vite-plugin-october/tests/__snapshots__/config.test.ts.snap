// Vitest Snapshot v1

exports[`does not read config the CONFIG_PATH_VITE if config is set to false 1`] = `
{
  "aliases": {
    "@": "assets",
  },
  "build_path": "build",
  "commands": {
    "artisan": [],
    "shell": [],
  },
  "configName": "artisan",
  "dev_server": {
    "cert": null,
    "enabled": true,
    "key": null,
    "ping_before_using_manifest": true,
    "ping_timeout": 1,
    "ping_url": null,
    "url": "http://localhost:3000",
  },
  "entrypoints": {
    "ignore": "/\\\\.(d\\\\.ts|json)$/",
    "paths": [
      "assets/scripts/main-artisan.ts",
    ],
    "ssr": "assets/scripts/ssr-artisan.ts",
  },
}
`;

exports[`reads config from the CONFIG_PATH_VITE environment variable 1`] = `
{
  "aliases": {
    "@": "assets",
  },
  "build_path": "build",
  "commands": {
    "artisan": [],
    "shell": [],
  },
  "configName": "default",
  "dev_server": {
    "cert": null,
    "enabled": true,
    "key": null,
    "ping_before_using_manifest": true,
    "ping_timeout": 1,
    "ping_url": null,
    "url": "http://localhost:3000",
  },
  "entrypoints": {
    "ignore": "/\\\\.(d\\\\.ts|json)$/",
    "paths": [
      "assets/scripts/main.ts",
    ],
    "ssr": "assets/scripts/ssr.ts",
  },
}
`;

exports[`reads config from the given option object 1`] = `
{
  "aliases": {
    "@": "assets",
  },
  "build_path": "build",
  "dev_server": {
    "url": "http://localhost:5173",
  },
  "entrypoints": {
    "paths": "assets/scripts/js",
  },
}
`;

exports[`reads config from the given path 1`] = `
{
  "aliases": {
    "@": "assets",
  },
  "build_path": "build",
  "commands": {
    "artisan": [],
    "shell": [],
  },
  "configName": "default",
  "dev_server": {
    "cert": null,
    "enabled": true,
    "key": null,
    "ping_before_using_manifest": true,
    "ping_timeout": 1,
    "ping_url": null,
    "url": "http://localhost:3000",
  },
  "entrypoints": {
    "ignore": "/\\\\.(d\\\\.ts|json)$/",
    "paths": [
      "assets/scripts/main.ts",
    ],
    "ssr": "assets/scripts/ssr.ts",
  },
}
`;

exports[`tries to read the specified configuration from php 1`] = `
{
  "aliases": {
    "@": "assets",
  },
  "build_path": "build",
  "commands": {
    "artisan": [],
    "shell": [],
  },
  "configName": "artisan",
  "dev_server": {
    "cert": null,
    "enabled": true,
    "key": null,
    "ping_before_using_manifest": true,
    "ping_timeout": 1,
    "ping_url": null,
    "url": "http://localhost:3000",
  },
  "entrypoints": {
    "ignore": "/\\\\.(d\\\\.ts|json)$/",
    "paths": [
      "assets/scripts/main-artisan.ts",
    ],
    "ssr": "assets/scripts/ssr-artisan.ts",
  },
}
`;
